File: DOHEE/Week1/성격 유형 검사하기.py
# 20분
def solution(survey, choices):
    answer = ''
    cake = {}
    type = ["R", "T", "C", "F", "J", "M", "A", "N"]

    for i in type:
        cake[i] = cake.get(i, 0)

    for i in range(len(survey)):
        if choices[i] > 4:
            cake[survey[i][1]] += choices[i] - 4
        elif choices[i] < 4:
            cake[survey[i][0]] += 4 - choices[i]

    answer += "T" if cake['R'] < cake['T'] else "R"
    answer += "F" if cake["C"] < cake["F"] else "C"
    answer += "M" if cake["J"] < cake["M"] else "J"
    answer += "N" if cake["A"] < cake["N"] else "A"
        
    return answer

File: DOHEE/Week1/성격 유형 검사하기.js
// 18분
function solution(survey, choices) {
    var answer = "";
    const result = { R: 0, T: 0, C: 0, F: 0, J: 0, M: 0, A: 0, N: 0 };
    survey.forEach((v, i) => {
        [a, b] = v.split("");
        if (choices[i] < 4) {
            result[a] += 4 - choices[i];
        } else if (choices[i] > 4) {
            result[b] += choices[i] - 4;
        }
    });
    answer += result["R"] < result["T"] ? "T" : "R";
    answer += result["C"] < result["F"] ? "F" : "C";
    answer += result["J"] < result["M"] ? "M" : "J";
    answer += result["A"] < result["N"] ? "N" : "A";
    return answer;
}
